src/app/static/src/tests/root/getters.test.ts

```diff
@@ -0,0 +1,78 @@
+import {mockModelCalibrateState, mockModelOptionsState, mockModelRunState, mockRootState} from "../mocks";
+import {getters} from "../../app/store/root/getters";
+import {Warning} from "../../app/generated";
+import {Steps} from "../../app/root";
+
+describe(`root getters`, () => {
+
+    const modelOptionWarnings: Warning[] = [
+        {text: "model option test", locations: ["model_options"]}
+    ]
+
+    const calibrateWarnings: Warning[] = [
+        {text: "model calibrate test", locations: ["model_options", "model_calibrate"]}
+    ]
+
+    const modelRunWarnings: Warning[] = [
+        {text: "model run test", locations: ["model_fit"]}
+    ]
+
+    const testState = () => {
+        return mockRootState({
+            modelOptions: mockModelOptionsState({warnings: modelOptionWarnings}),
+            modelRun: mockModelRunState({warnings: modelRunWarnings}),
+            modelCalibrate: mockModelCalibrateState({warnings: calibrateWarnings})
+        })
+    }
+
+    it(`can get modelRun warnings when on model run step`, () => {
+        const rootState = testState()
+
+        const warn = getters.warnings(rootState, null, testState() as any, null)
+        const result = warn(Steps.modelRun).modelRun
```

Comment:
Personally I'd find the tests easier to read if variables where renamed as follows, but not a huge deal:
```suggestion
        const getter = getters.warnings(rootState, null, testState() as any, null)
        const warnings = warn(Steps.modelRun).modelRun
```